File: apps/ios/BankCore/Design/Components.swift
import SwiftUI
import UIKit

struct Wordmark: View {
    var size: CGFloat = 32

    var body: some View {
        HStack(spacing: 0) {
            Text("Bank")
                .foregroundStyle(Palette.ivory)
            Text("Core")
                .foregroundStyle(Palette.gold)
        }
        .font(.system(size: size, weight: .semibold))
        .tracking(-0.4)
    }
}

struct BrandMark: View {
    var size: CGFloat = 48

    var body: some View {
        Image(systemName: "shield")
            .font(.system(size: size * 0.42, weight: .medium))
            .foregroundStyle(Palette.gold)
            .frame(width: size, height: size)
            .overlay(
                RoundedRectangle(cornerRadius: size * 0.22, style: .continuous)
                    .stroke(Palette.gold, lineWidth: max(1.2, size * 0.035))
            )
    }
}

enum AppVersion {
    static var marketing: String {
        Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? "1.10.25"
    }

    static var line: String { "version \(marketing)" }
}

struct VersionLabel: View {
    var body: some View {
        Text(AppVersion.line)
            .font(.system(size: 11, weight: .medium, design: .monospaced))
            .foregroundStyle(Palette.mute)
            .tracking(0.6)
    }
}

struct CarbonCard<Content: View>: View {
    var padding: CGFloat = 16
    @ViewBuilder var content: Content

    var body: some View {
        content
            .padding(padding)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(Palette.card)
            .overlay(
                RoundedRectangle(cornerRadius: 12, style: .continuous)
                    .stroke(Palette.line, lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
    }
}

struct GoldButton: View {
    let title: String
    var systemImage: String? = nil
    var isLoading: Bool = false
    var enabled: Bool = true
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 8) {
                if isLoading {
                    ProgressView()
                        .tint(Palette.ink)
                } else if let systemImage {
                    Image(systemName: systemImage)
                }
                Text(title.uppercased())
                    .font(TypeScale.cta)
                    .tracking(0.8)
            }
            .frame(maxWidth: .infinity)
            .padding(.vertical, 14)
            .foregroundStyle(Palette.ink)
            .background(enabled && !isLoading ? Palette.gold : Palette.gold.opacity(0.35))
            .clipShape(RoundedRectangle(cornerRadius: 8, style: .continuous))
        }
        .buttonStyle(.plain)
        .disabled(!enabled || isLoading)
    }
}

struct SecondaryButton: View {
    let title: String
    var systemImage: String? = nil
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 8) {
                if let systemImage {
                    Image(systemName: systemImage)
                }
                Text(title)
                    .font(TypeScale.cta)
            }
            .frame(maxWidth: .infinity)
            .padding(.vertical, 13)
            .foregroundStyle(Palette.ivory)
            .overlay(
                RoundedRectangle(cornerRadius: 8, style: .continuous)
                    .stroke(Palette.line, lineWidth: 1)
            )
        }
        .buttonStyle(.plain)
    }
}

struct FieldLabel: View {
    let text: String
    var body: some View {
        Text(text)
            .font(TypeScale.label)
            .foregroundStyle(Palette.mute)
    }
}

struct CarbonField: View {
    let label: String
    @Binding var text: String
    var placeholder: String = ""
    var keyboard: UIKeyboardType = .default
    var isSecure: Bool = false
    var mono: Bool = false

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            FieldLabel(text: label)
            Group {
                if isSecure {
                    SecureField(placeholder, text: $text)
                } else {
                    TextField(placeholder, text: $text)
                        .keyboardType(keyboard)
                }
            }
            .font(mono ? .system(size: 15, design: .monospaced) : TypeScale.body)
            .foregroundStyle(Palette.ivory)
            .textInputAutocapitalization(.never)
            .autocorrectionDisabled()
            .padding(.horizontal, 14)
            .padding(.vertical, 13)
            .background(Palette.input)
            .overlay(
                RoundedRectangle(cornerRadius: 8, style: .continuous)
                    .stroke(Palette.line, lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: 8, style: .continuous))
        }
    }
}

struct DemoBanner: View {
    var body: some View {
        Text("Ambiente de portfólio — login, Pix, extrato e comprovante no ledger interno")
            .font(TypeScale.micro)
            .foregroundStyle(Palette.gold)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
    }
}

struct AmountText: View {
    let value: Double
    var credit: Bool? = nil
    var size: Font = TypeScale.amount

    var body: some View {
        Text(label)
            .font(size)
            .foregroundStyle(color)
            .monospacedDigit()
    }

    private var label: String {
        if let credit {
            return Money.signed(value, credit: credit)
        }
        return Money.reais(value)
    }

    private var color: Color {
        if credit == false { return Palette.debit }
        return Palette.ivory
    }
}

struct TransactionRow: View {
    let transaction: LedgerTransaction

    var body: some View {
        HStack(spacing: 12) {
            ZStack {
                RoundedRectangle(cornerRadius: 8, style: .continuous)
                    .fill(transaction.isCredit ? Palette.gold.opacity(0.12) : Palette.debit.opacity(0.12))
                    .overlay(
                        RoundedRectangle(cornerRadius: 8, style: .continuous)
                            .stroke(transaction.isCredit ? Palette.gold.opacity(0.28) : Palette.debit.opacity(0.28), lineWidth: 1)
                    )
                Image(systemName: transaction.isCredit ? "arrow.down.left" : "arrow.up.right")
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundStyle(transaction.isCredit ? Palette.gold : Palette.debit)
            }
            .frame(width: 36, height: 36)

            VStack(alignment: .leading, spacing: 3) {
                Text(transaction.title)
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundStyle(Palette.ivory)
                    .lineLimit(1)
                Text("\(BankDate.short(transaction.createdAt)) · \(transaction.idempotencyKey.prefix(12))…")
                    .font(.system(size: 11, design: .monospaced))
                    .foregroundStyle(Palette.mute)
            }

            Spacer(minLength: 8)

            AmountText(value: transaction.amountReais, credit: transaction.isCredit)
        }
        .padding(.vertical, 10)
    }
}

struct StatusBadge: View {
    let text: String
    var body: some View {
        Text(text.uppercased())
            .font(.system(size: 10, weight: .semibold))
            .tracking(0.6)
            .foregroundStyle(Palette.status)
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .overlay(
                Capsule()
                    .stroke(Palette.status.opacity(0.35), lineWidth: 1)
            )
            .background(Palette.status.opacity(0.12), in: Capsule())
    }
}

struct SimulatedBadge: View {
    var body: some View {
        Text("SIMULADO")
            .font(.system(size: 9, weight: .semibold))
            .tracking(0.6)
            .foregroundStyle(Palette.gold)
            .padding(.horizontal, 7)
            .padding(.vertical, 3)
            .overlay(
                Capsule().stroke(Palette.gold.opacity(0.45), lineWidth: 1)
            )
    }
}

struct InitialsAvatar: View {
    let initials: String
    var size: CGFloat = 44

    var body: some View {
        Text(initials)
            .font(.system(size: size * 0.36, weight: .semibold))
            .foregroundStyle(Palette.gold)
            .frame(width: size, height: size)
            .background(Palette.gold.opacity(0.12))
            .overlay(
                Circle().stroke(Palette.gold.opacity(0.55), lineWidth: 1)
            )
            .clipShape(Circle())
    }
}

struct ShortcutButton: View {
    let title: String
    let systemImage: String
    var action: () -> Void

    var body: some View {
        Button(action: action) {
            VStack(spacing: 8) {
                ZStack {
                    Circle()
                        .fill(Palette.gold.opacity(0.12))
                    Image(systemName: systemImage)
                        .font(.system(size: 18, weight: .medium))
                        .foregroundStyle(Palette.gold)
                }
                .frame(width: 56, height: 56)
                Text(title)
                    .font(.system(size: 11, weight: .medium))
                    .foregroundStyle(Palette.ivory)
                    .multilineTextAlignment(.center)
                    .lineLimit(2)
                    .frame(height: 28, alignment: .top)
            }
            .frame(maxWidth: .infinity)
        }
        .buttonStyle(.plain)
    }
}

struct LimitBar: View {
    let ratio: Double

    var body: some View {
        GeometryReader { geo in
            ZStack(alignment: .leading) {
                Capsule().fill(Palette.line)
                Capsule()
                    .fill(Palette.gold)
                    .frame(width: max(8, geo.size.width * ratio))
            }
        }
        .frame(height: 6)
    }
}

struct SectionTitle: View {
    let text: String
    var body: some View {
        Text(text)
            .font(.system(size: 20, weight: .semibold))
            .foregroundStyle(Palette.ivory)
    }
}

struct SettingsRow: View {
    let icon: String
    let title: String
    var subtitle: String? = nil
    var action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(alignment: .top, spacing: 12) {
                Image(systemName: icon)
                    .font(.system(size: 18, weight: .medium))
                    .foregroundStyle(Palette.gold)
                    .frame(width: 28, alignment: .center)
                    .padding(.top, 2)
                VStack(alignment: .leading, spacing: 3) {
                    Text(title)
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundStyle(Palette.ivory)
                    if let subtitle {
                        Text(subtitle)
                            .font(TypeScale.label)
                            .foregroundStyle(Palette.mute)
                            .fixedSize(horizontal: false, vertical: true)
                    }
                }
                Spacer()
                Image(systemName: "chevron.right")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundStyle(Palette.mute)
            }
            .padding(.vertical, 12)
        }
        .buttonStyle(.plain)
    }
}

struct PlasticCard: View {
    let card: MockCard

    var body: some View {
        VStack(alignment: .leading, spacing: 14) {
            HStack {
                HStack(spacing: 0) {
                    Text("Bank").foregroundStyle(.white)
                    Text("Core").foregroundStyle(Palette.gold)
                }
                .font(.system(size: 13, weight: .semibold))
                Spacer()
                Text(card.name.uppercased())
                    .font(.system(size: 10, weight: .semibold))
                    .tracking(0.8)
                    .foregroundStyle(Palette.gold)
            }
            Text(Money.reais(card.invoice))
                .font(.system(size: 26, weight: .semibold).monospacedDigit())
                .foregroundStyle(.white)
            HStack {
                VStack(alignment: .leading, spacing: 2) {
                    Text("Vencimento").font(TypeScale.micro).foregroundStyle(Palette.mute)
                    Text(card.dueLabel).font(.system(size: 12, weight: .medium)).foregroundStyle(.white)
                }
                Spacer()
                VStack(alignment: .trailing, spacing: 2) {
                    Text("Compras").font(TypeScale.micro).foregroundStyle(Palette.mute)
                    Text(card.period).font(.system(size: 12, weight: .medium)).foregroundStyle(.white)
                }
            }
            HStack {
                Text("•••• \(card.last4)")
                    .font(.system(size: 13, design: .monospaced))
                    .foregroundStyle(Palette.ivory.opacity(0.8))
                Spacer()
                Text(card.brand.uppercased())
                    .font(.system(size: 11, weight: .bold))
                    .foregroundStyle(Palette.mute)
            }
        }
        .padding(18)
        .frame(maxWidth: .infinity, minHeight: 176, alignment: .leading)
        .background(gradient)
        .overlay(
            RoundedRectangle(cornerRadius: 16, style: .continuous)
                .stroke(Palette.gold.opacity(0.35), lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous))
    }

    private var gradient: LinearGradient {
        switch card.theme {
        case .black:
            return LinearGradient(colors: [Color(hex: "#1A1A1C"), Color(hex: "#0D0D0E")], startPoint: .topLeading, endPoint: .bottomTrailing)
        case .platinum:
            return LinearGradient(colors: [Color(hex: "#2A2A2E"), Color(hex: "#141416")], startPoint: .topLeading, endPoint: .bottomTrailing)
        case .virtual:
            return LinearGradient(colors: [Color(hex: "#161618"), Color(hex: "#0B0B0C")], startPoint: .topLeading, endPoint: .bottomTrailing)
        }
    }
}

File: apps/android/app/src/main/java/br/vortex/bankcore/ui/components/Components.kt
package br.vortex.bankcore.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.KeyboardArrowRight
import androidx.compose.material.icons.outlined.NorthEast
import androidx.compose.material.icons.outlined.SouthWest
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.StrokeJoin
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import br.vortex.bankcore.data.LedgerTransaction
import br.vortex.bankcore.data.MockCard
import br.vortex.bankcore.ui.theme.Palette
import br.vortex.bankcore.ui.theme.hex
import br.vortex.bankcore.util.BankDate
import br.vortex.bankcore.util.Money

@Composable
fun Wordmark(size: Int = 32, modifier: Modifier = Modifier) {
    val p = Palette
    Row(modifier = modifier) {
        Text(
            "Bank",
            color = p.ivory,
            fontSize = size.sp,
            fontWeight = FontWeight.SemiBold,
            letterSpacing = (-0.4).sp,
        )
        Text(
            "Core",
            color = p.gold,
            fontSize = size.sp,
            fontWeight = FontWeight.SemiBold,
            letterSpacing = (-0.4).sp,
        )
    }
}

@Composable
fun BrandMark(size: Dp = 48.dp, modifier: Modifier = Modifier) {
    val gold = Palette.gold
    Canvas(modifier = modifier.size(size)) {
        val stroke = maxOf(1.2.dp.toPx(), size.toPx() * 0.035f)
        val inset = stroke / 2
        val radius = size.toPx() * 0.22f
        drawRoundRect(
            color = gold,
            topLeft = Offset(inset, inset),
            size = Size(this.size.width - stroke, this.size.height - stroke),
            cornerRadius = androidx.compose.ui.geometry.CornerRadius(radius, radius),
            style = Stroke(width = stroke),
        )
        val cx = this.size.width / 2
        val cy = this.size.height / 2 + this.size.height * 0.02f
        val w = this.size.width * 0.38f
        val h = this.size.height * 0.42f
        val path = Path().apply {
            val left = cx - w / 2
            val right = cx + w / 2
            val top = cy - h * 0.46f
            val bottom = cy + h * 0.50f
            val r = w * 0.22f
            val waist = cy + h * 0.08f
            moveTo(left + r, top)
            quadraticTo(left, top, left, top + r)
            lineTo(left, waist)
            quadraticTo(left + w * 0.06f, bottom - h * 0.12f, cx, bottom)
            quadraticTo(right - w * 0.06f, bottom - h * 0.12f, right, waist)
            lineTo(right, top + r)
            quadraticTo(right, top, right - r, top)
            close()
        }
        drawPath(
            path,
            color = gold,
            style = Stroke(width = stroke, cap = StrokeCap.Round, join = StrokeJoin.Round),
        )
    }
}

@Composable
fun VersionLabel(modifier: Modifier = Modifier) {
    Text(
        "version 1.10.25",
        modifier = modifier,
        fontSize = 11.sp,
        fontWeight = FontWeight.Medium,
        fontFamily = FontFamily.Monospace,
        color = Palette.mute,
        letterSpacing = 0.6.sp,
    )
}

@Composable
fun CarbonCard(
    modifier: Modifier = Modifier,
    padding: Dp = 16.dp,
    content: @Composable ColumnScope.() -> Unit,
) {
    val p = Palette
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(p.card)
            .border(1.dp, p.line, RoundedCornerShape(12.dp))
            .padding(padding),
        content = content,
    )
}

@Composable
fun GoldButton(
    title: String,
    modifier: Modifier = Modifier,
    icon: ImageVector? = null,
    isLoading: Boolean = false,
    enabled: Boolean = true,
    onClick: () -> Unit,
) {
    val p = Palette
    val active = enabled && !isLoading
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(8.dp))
            .background(if (active) p.gold else p.gold.copy(alpha = 0.35f))
            .clickable(enabled = active, onClick = onClick)
            .padding(vertical = 14.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        if (isLoading) {
            CircularProgressIndicator(
                modifier = Modifier.size(16.dp),
                color = p.onGold,
                strokeWidth = 2.dp,
            )
            Spacer(Modifier.width(8.dp))
        } else if (icon != null) {
            Icon(icon, contentDescription = null, tint = p.onGold, modifier = Modifier.size(16.dp))
            Spacer(Modifier.width(8.dp))
        }
        Text(
            title.uppercase(),
            color = p.onGold,
            fontSize = 13.sp,
            fontWeight = FontWeight.SemiBold,
            letterSpacing = 0.8.sp,
        )
    }
}

@Composable
fun SecondaryButton(
    title: String,
    modifier: Modifier = Modifier,
    icon: ImageVector? = null,
    onClick: () -> Unit,
) {
    val p = Palette
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(8.dp))
            .border(1.dp, p.line, RoundedCornerShape(8.dp))
            .clickable(onClick = onClick)
            .padding(vertical = 13.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        if (icon != null) {
            Icon(icon, contentDescription = null, tint = p.ivory, modifier = Modifier.size(16.dp))
            Spacer(Modifier.width(8.dp))
        }
        Text(title, color = p.ivory, fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
    }
}

@Composable
fun CarbonField(
    label: String,
    value: String,
    onValueChange: (String) -> Unit,
    modifier: Modifier = Modifier,
    placeholder: String = "",
    keyboardType: KeyboardType = KeyboardType.Text,
    isSecure: Boolean = false,
    mono: Boolean = false,
) {
    val p = Palette
    Column(modifier = modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(label, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = p.mute)
        OutlinedTextField(
            value = value,
            onValueChange = onValueChange,
            modifier = Modifier.fillMaxWidth(),
            placeholder = { Text(placeholder, color = p.mute) },
            singleLine = true,
            visualTransformation = if (isSecure) PasswordVisualTransformation() else VisualTransformation.None,
            keyboardOptions = KeyboardOptions(keyboardType = keyboardType),
            textStyle = androidx.compose.ui.text.TextStyle(
                fontSize = 15.sp,
                color = p.ivory,
                fontFamily = if (mono) FontFamily.Monospace else FontFamily.SansSerif,
            ),
            colors = OutlinedTextFieldDefaults.colors(
                focusedBorderColor = p.gold,
                unfocusedBorderColor = p.line,
                focusedContainerColor = p.input,
                unfocusedContainerColor = p.input,
                cursorColor = p.gold,
                focusedTextColor = p.ivory,
                unfocusedTextColor = p.ivory,
            ),
            shape = RoundedCornerShape(8.dp),
        )
    }
}

@Composable
fun DemoBanner() {
    Text(
        "Ambiente de portfólio — login, Pix, extrato e comprovante no ledger interno",
        fontSize = 11.sp,
        fontWeight = FontWeight.Medium,
        color = Palette.gold,
        textAlign = TextAlign.Center,
        modifier = Modifier.fillMaxWidth(),
    )
}

@Composable
fun AmountText(value: Double, credit: Boolean? = null, size: Int = 15) {
    val p = Palette
    val label = if (credit != null) Money.signed(value, credit) else Money.reais(value)
    val color = if (credit == false) p.debit else p.ivory
    Text(
        label,
        fontSize = size.sp,
        fontWeight = FontWeight.SemiBold,
        fontFamily = FontFamily.Monospace,
        color = color,
    )
}

@Composable
fun TransactionRow(transaction: LedgerTransaction, onClick: () -> Unit) {
    val p = Palette
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(vertical = 10.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        val tint = if (transaction.isCredit) p.gold else p.debit
        Box(
            modifier = Modifier
                .size(36.dp)
                .clip(RoundedCornerShape(8.dp))
                .background(tint.copy(alpha = 0.12f))
                .border(1.dp, tint.copy(alpha = 0.28f), RoundedCornerShape(8.dp)),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                if (transaction.isCredit) Icons.Outlined.SouthWest else Icons.Outlined.NorthEast,
                contentDescription = null,
                tint = tint,
                modifier = Modifier.size(14.dp),
            )
        }
        Spacer(Modifier.width(12.dp))
        Column(modifier = Modifier.weight(1f)) {
            Text(
                transaction.title,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                color = p.ivory,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            Text(
                "${BankDate.short(transaction.createdAt)} · ${transaction.idempotencyKey.take(12)}…",
                fontSize = 11.sp,
                fontFamily = FontFamily.Monospace,
                color = p.mute,
            )
        }
        Spacer(Modifier.width(8.dp))
        AmountText(transaction.amountReais, credit = transaction.isCredit)
    }
}

@Composable
fun StatusBadge(text: String) {
    val p = Palette
    Text(
        text.uppercase(),
        fontSize = 10.sp,
        fontWeight = FontWeight.SemiBold,
        letterSpacing = 0.6.sp,
        color = p.status,
        modifier = Modifier
            .clip(RoundedCornerShape(50))
            .background(p.status.copy(alpha = 0.12f))
            .border(1.dp, p.status.copy(alpha = 0.35f), RoundedCornerShape(50))
            .padding(horizontal = 8.dp, vertical = 4.dp),
    )
}

@Composable
fun SimulatedBadge() {
    val p = Palette
    Text(
        "SIMULADO",
        fontSize = 9.sp,
        fontWeight = FontWeight.SemiBold,
        letterSpacing = 0.6.sp,
        color = p.gold,
        modifier = Modifier
            .border(1.dp, p.gold.copy(alpha = 0.45f), RoundedCornerShape(50))
            .padding(horizontal = 7.dp, vertical = 3.dp),
    )
}

@Composable
fun InitialsAvatar(initials: String, size: Dp = 44.dp) {
    val p = Palette
    Box(
        modifier = Modifier
            .size(size)
            .clip(CircleShape)
            .background(p.gold.copy(alpha = 0.12f))
            .border(1.dp, p.gold.copy(alpha = 0.55f), CircleShape),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            initials,
            fontSize = (size.value * 0.36f).sp,
            fontWeight = FontWeight.SemiBold,
            color = p.gold,
        )
    }
}

@Composable
fun ShortcutButton(title: String, icon: ImageVector, onClick: () -> Unit) {
    val p = Palette
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(vertical = 4.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Box(
            modifier = Modifier
                .size(56.dp)
                .clip(CircleShape)
                .background(p.gold.copy(alpha = 0.12f)),
            contentAlignment = Alignment.Center,
        ) {
            Icon(icon, contentDescription = title, tint = p.gold, modifier = Modifier.size(22.dp))
        }
        Spacer(Modifier.height(8.dp))
        Text(
            title,
            fontSize = 11.sp,
            fontWeight = FontWeight.Medium,
            color = p.ivory,
            textAlign = TextAlign.Center,
            maxLines = 2,
            modifier = Modifier.height(28.dp),
        )
    }
}

@Composable
fun LimitBar(ratio: Double) {
    val p = Palette
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(6.dp)
            .clip(RoundedCornerShape(50))
            .background(p.line),
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth(ratio.toFloat().coerceIn(0.04f, 1f))
                .height(6.dp)
                .clip(RoundedCornerShape(50))
                .background(p.gold),
        )
    }
}

@Composable
fun SectionTitle(text: String) {
    Text(text, fontSize = 20.sp, fontWeight = FontWeight.SemiBold, color = Palette.ivory)
}

@Composable
fun SettingsRow(
    icon: ImageVector,
    title: String,
    subtitle: String? = null,
    onClick: () -> Unit,
) {
    val p = Palette
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(vertical = 12.dp),
        verticalAlignment = Alignment.Top,
    ) {
        Icon(icon, contentDescription = null, tint = p.gold, modifier = Modifier.size(22.dp).padding(top = 2.dp))
        Spacer(Modifier.width(12.dp))
        Column(modifier = Modifier.weight(1f)) {
            Text(title, fontSize = 16.sp, fontWeight = FontWeight.SemiBold, color = p.ivory)
            if (subtitle != null) {
                Text(subtitle, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = p.mute)
            }
        }
        Icon(Icons.AutoMirrored.Outlined.KeyboardArrowRight, contentDescription = null, tint = p.mute, modifier = Modifier.size(16.dp))
    }
}

@Composable
fun PlasticCard(card: MockCard, modifier: Modifier = Modifier) {
    val p = Palette
    val colors = when (card.theme) {
        MockCard.Theme.BLACK -> listOf(hex("#1A1A1C"), hex("#0D0D0E"))
        MockCard.Theme.PLATINUM -> listOf(hex("#2A2A2E"), hex("#141416"))
        MockCard.Theme.VIRTUAL -> listOf(hex("#161618"), hex("#0B0B0C"))
    }
    Column(
        modifier = modifier
            .fillMaxWidth()
            .height(176.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(Brush.linearGradient(colors))
            .border(1.dp, p.gold.copy(alpha = 0.35f), RoundedCornerShape(16.dp))
            .padding(18.dp),
        verticalArrangement = Arrangement.SpaceBetween,
    ) {
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Row {
                Text("Bank", color = Color.White, fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
                Text("Core", color = p.gold, fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
            }
            Text(
                card.name.uppercase(),
                fontSize = 10.sp,
                fontWeight = FontWeight.SemiBold,
                letterSpacing = 0.8.sp,
                color = p.gold,
            )
        }
        Text(
            Money.reais(card.invoice),
            fontSize = 26.sp,
            fontWeight = FontWeight.SemiBold,
            fontFamily = FontFamily.Monospace,
            color = Color.White,
        )
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Column {
                Text("Vencimento", fontSize = 11.sp, color = p.mute)
                Text(card.dueLabel, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = Color.White)
            }
            Column(horizontalAlignment = Alignment.End) {
                Text("Compras", fontSize = 11.sp, color = p.mute)
                Text(card.period, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = Color.White)
            }
        }
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text("•••• ${card.last4}", fontSize = 13.sp, fontFamily = FontFamily.Monospace, color = Color.White.copy(alpha = 0.8f))
            Text(card.brand.uppercase(), fontSize = 11.sp, fontWeight = FontWeight.Bold, color = p.mute)
        }
    }
}

@Composable
fun Hairline() {
    Box(Modifier.fillMaxWidth().height(1.dp).background(Palette.line))
}
